qemu/tests/timerdevice_boot.py

```diff
@@ -128,26 +162,30 @@ def verify_guest_clock_source(session, expected):
     time_format = params["time_format"]
     timerdevice_drift_threshold = params.get("timerdevice_drift_threshold", 3)
 
-    error.context("Check the system time on guest and host", logging.info)
-    (host_time, guest_time) = utils_test.get_time(session, time_command,
-                                                  time_filter_re, time_format)
-    drift = abs(float(host_time) - float(guest_time))
-    if drift > timerdevice_drift_threshold:
-        raise error.TestFail("The guest's system time is different with"
-                             " host's. Host time: '%s', guest time:"
-                             " '%s'" % (host_time, guest_time))
-
-    get_hw_time_cmd = params.get("get_hw_time_cmd")
-    if get_hw_time_cmd:
-        error.context(
-            "Check the hardware time on guest and host", logging.info)
-        host_time = utils.system_output(get_hw_time_cmd)
-        guest_time = session.cmd(get_hw_time_cmd)
-        drift = abs(float(host_time) - float(guest_time))
-        if drift > timerdevice_drift_threshold:
-            raise error.TestFail("The guest's hardware time is different with"
-                                 " host's. Host time: '%s', guest time:"
-                                 " '%s'" % (host_time, guest_time))
+    if params["rtc_base"] in ("utc", "localtime"):
+        verify_guest_system_time(time_command, time_filter_re, time_format)
+        verify_guest_hw_time(time_command, time_filter_re, time_format)
+    else:
+        if params["os_type"] == "linux":
+            chronyd_active = session.cmd_output_safe("service chronyd status"
+                                                     "| grep 'active (running)'")
+            # if chronyd service is enabled
+            # Mask chronyd service, shutdown the guest
+            # Boot guest again
+            if chronyd_active:
```

Comment:
I have tried above command in my fedora host, it will give an output 'Redirecting to /bin/systemctl status  chronyd.service'(I supposed that it should act same way for RHEL7) even it can't match the string I grep, therefore this judgement will be satisfied even chronyd service not running.